course_dashboard/problem_stats/question_monitors.py

```diff
@@ -169,13 +170,41 @@ def format_student_answers(self):
               u"[u'choice_1',u'choice_2']" -> (2, 3)
         The first choice beeing 1 and not 0.
         """
-        student_answers = {}
+        student_answers = defaultdict(int)
         for answer, value in self.student_answers.iteritems():
-            choices = ast.literal_eval(answer)
-            choices = [int(choice[-1:]) + 1 for choice in choices]
-            student_answers[tuple(choices)] = value
+            # When the problem is changed from a multiple choice question (with one
+            # possible answer) to a question problem (with multiple allowed
+            # answers) the ChoiceQuestionMonitor has to parse answers formatted
+            # differently.
+            choices = self.parse_answer(answer) or self.parse_answer("['" + answer + "']") or ()
```

Comment:
Would it be simpler to understand if we move the answer reformatting and the corresponding doc, directly into the `parse_answer()` method ? This way, we can rely on `choices = self.parse_answer(answer)` no matter what you pass in as answer.